### PR TITLE
Add a new item called `type` to `hdd` object

### PR DESCRIPTION
**Pull request by sefeng211**
This is the schema change for https://bugzilla.mozilla.org/show_bug.cgi?id=1533861

Checklist for reviewer:

- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] Trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional)

For glean changes:
- [ ] Update `include/glean/CHANGELOG.md`

### CODE REVIEW

File: templates/include/telemetry/environment.1.schema.json [hunk 1]
```diff
@@ -656,6 +656,12 @@
                     "string",
                     "null"
                   ]
+                },
+                "type": {
+                  "type": [
+                    "string",
+                    "null"
+                  ]
```
Comment (acmiyaguchi) [inline templates/include/telemetry/environment.1.schema.json]: It looks like this can be an enumerated type, i.e.

```json
enum: ["SSD", "HDD"]
```

Comment (sefeng211) [reply templates/include/telemetry/environment.1.schema.json]: Thanks for pointing this out! Updated the PR. 

I kept the `type` as 
```
"type": [
  "string",
  "null"
]
```

Would `null` still work?

Comment (acmiyaguchi) [reply templates/include/telemetry/environment.1.schema.json]: `null` values should still be valid because nothing has changed other than limiting the the values of `string` to the values contained in `enum`. If you aren't sure, I would encourage you to explicitly write another validation test that includes null (and missing!) values for the new column you've added.

Comment (sefeng211) [reply templates/include/telemetry/environment.1.schema.json]: Thanks! Added two more tests, for null and empty string.

File: validation/telemetry/main.4.hdd_type_null.fail.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      "vendor": "Mozilla",
+      "platformVersion": "62.0a1",
+      "xpcomAbi": "x86_64-gcc3"
+    },
+    "partner": {},
+    "settings": {},
+    "system": {
+      "hdd": {
+        "profile": {
+          "type": null
```
Comment (acmiyaguchi) [inline validation/telemetry/main.4.hdd_type_null.fail.json]: I actually didn't know this would fail. It looks like there needs to be a slight modification for this to work properly.

File: templates/include/telemetry/environment.1.schema.json [hunk 3]
```diff
@@ -656,6 +656,16 @@
...
                   ]
+                },
+                "type": {
+                  "type": [
+                    "string",
+                    "null"
+                  ],
+                  "enum": [
+                    "SSD",
+                    "HDD"
```
Comment (acmiyaguchi) [inline templates/include/telemetry/environment.1.schema.json]:
```suggestion
                    "HDD",
                    null
```

This should allow null values for this field, while also limiting the values to the enumerated set.

File: validation/telemetry/main.4.hdd_type_null.fail.json (hunk 2 above)
Comment (sefeng211) [reply validation/telemetry/main.4.hdd_type_null.fail.json]: Ah yep, `null` should pass. It is fixed now

File: templates/include/telemetry/environment.1.schema.json (hunk 3 above)
Comment (sefeng211) [reply templates/include/telemetry/environment.1.schema.json]: PR updated!

**Comment by sefeng211** [comment]
The patch is reached `mozilla-central`, do you mind look it again @acmiyaguchi ? Thanks!
